nemo/core/classes/modelPT.py

```diff
@@ -1735,6 +1744,75 @@ def update_save_restore_connector(cls, save_restore_connector):
         else:
             setattr(cls, '_save_restore_connector', save_restore_connector)
 
+    def _setup_chakra_profiling(self):
+        """ Enables chakra profiling
+            To use, add the following optoins to the model config:
+            ## Chakra profiling options
+            chakra_profile:
+                enabled: False
+                start_step: 2  # Global batch to start profiling
+                end_step: 2 # Global batch to end profiling
+                warmup_steps: 0  # Global batch to start profiling
+                active_steps: 1  # Global batch to start profiling
+                trace_dir: None # Path to store the profile output file
+        """
+        if self.cfg.get('chakra_profile', None) is not None:
+            if self.cfg.chakra_profile.get('enabled', False):
+                self._chakra_profile_enabled = True
+                self._chakra_profile_start_step = self.cfg.chakra_profile.get('start_step', 0)
+                self._chakra_profile_end_step = self.cfg.chakra_profile.get('end_step', 0)
+
+                if self.cfg.chakra_profile.get('trace_dir', None) is None:
+                    raise ValueError(
+                        f'chakra profile output path ({self._memory_profile_output_path}) is not set.'
```

Comment:
Seems this hasn't been assigned yet? Also can we do '_chakra_output_path' for consistency, since 'memory_profile' prefix is used for all the Nsys attributes.

Reply:
Resolved in the latest commit, thank you for the reminder